src/control/shaperecognizer/ShapeRecognizer.cpp

```diff
@@ -297,22 +297,33 @@ auto ShapeRecognizer::recognizePatterns(Stroke* stroke) -> ShapeRecognizerResult
 
         if (n == 1)  // current stroke is a line
         {
+            bool aligned = false;
             if (fabs(rs->angle) < SLANT_TOLERANCE)  // nearly horizontal
             {
                 rs->angle = 0.0;
                 rs->y1 = rs->y2 = rs->ycenter;
+                aligned = true;
             }
             if (fabs(rs->angle) > M_PI / 2 - SLANT_TOLERANCE)  // nearly vertical
```

Comment:
This should be an `else if`. You could also merge the logic for adding the point with this logic but that is not really necessary.